### PR TITLE
Use Java 17 on CI.

### PR DESCRIPTION
**Pull request by colinrtwhite**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -17,7 +17,8 @@ jobs:
             - uses: actions/setup-java@v2
               with:
                   distribution: 'adopt'
-                  java-version: 11
+                  java-version: 17
```
Comment (Goooler) [inline .github/workflows/ci.yml:20]: Suggesting use `zulu` instead of `adopt`,
https://api.adoptopenjdk.net/

Comment (colinrtwhite) [reply .github/workflows/ci.yml:20]: @Goooler What's the benefit? I thought `zulu` was best for M1 Macs, though I don't think the Github Mac runners are M1.

Comment (Goooler) [reply .github/workflows/ci.yml:20]: As jake recommending this.

Comment (colinrtwhite) [reply .github/workflows/ci.yml:20]: Sounds good to me :)

**Pull request merged**